src/swagger-config.yaml

```diff
@@ -9,46 +9,84 @@ info:
     name: Apache 2.0
     url: 'http://www.apache.org/licenses/LICENSE-2.0.html'
 paths:
-  '/components/{componentId}':
+  '/component-library':
     get:
       tags:
         - developers
-      parameters:
-        - in: path
-          name: componentId
-          required: true
-          schema:
-            type: string
-          example: Modelica.Electrical.Analog.Basic.Inductor
       responses:
         '200':
-          description: Component with specified componentId
+          description: Hierarchical list of all components
           content:
             application/json:
               schema:
-                $ref: '#/components/schemas/Component'
-  '/components':
-    get:
-      tags:
-        - developers
-      responses:
-        '200':
-          description: List of all components
-          content:
-            application/json:
-              schema:
-                type: array
-                items:
-                  $ref: '#/components/schemas/Component'
+                $ref: '#/components/schemas/Collection'
+
 components:
   schemas:
+    Collection:
+      type: object
+      properties:
+        id:
+          type: string
+        displayLabel:
+          type: string
+        nodeType:
+          type: string
+        svgPath:
+          type: string
+        children:
+          type: array
+          items:
+            anyOf:
+              - $ref: '#/components/schemas/Collection'
+              - $ref: '#/components/schemas/Component'
+      example:
+        id: Modelica.Electrical
+        displayLabel: Electrical
+        nodeType: collection
+        svgPath: https://hostname:port/assets/Modelica.Electrical.svg
+        children:
+          - id: Modelica.Electrical.Analog
+            displayLabel: Analog
+            nodeType: collection
```

Comment:
I suggest instead of inventing new types we should use Modelica restriction here. Could be useful!

Reply:
What is a Modelica restriction?

Reply:
It defines the type of the class/node. See section [4.6 Specialized Classes](https://specification.modelica.org/master/class-predefined-types-and-declarations.html#specialized-classes). For example, instead of saying `collection` we could say `package`.

Reply:
Ok, while in general, I would like to avoid using anything that's too Modelica specific, `package` is still general enough, so will refactor to use it instead of `collection`.

What would you suggest we use instead of `nodeType: component` ?

Reply:
I would prefer to call them model and connector but they could be other restriction type as well.

If we want to keep it generic then I agree that we can use collection and component. Maybe for now we use generic types.